src/game/Tactical/Soldier_Ani.cc

```diff
@@ -659,8 +657,8 @@ BOOLEAN AdjustToNextAnimationFrame( SOLDIERTYPE *pSoldier )
 					else if (pSoldier->bDoBurst == 1)
 					{
 						// CHECK FOR GUN JAM
-						bWeaponJammed = CheckForGunJam( pSoldier );
-						if ( bWeaponJammed == TRUE )
+						if (auto const weaponJammed = CheckForGunJam(pSoldier);
+						    weaponJammed == FireWeaponResult::JAMMED)
```

Comment:
Bljak is this ugly, please leave the first statement out of the test.

Reply:
Really? I kind of like it because it limits the scope of the variable to exactly the if statement.

Reply:
It's worse for readability and bug prone. So much in fact there are several standards warning against it (copying from Sonar):

- MISRA C:2004, 13.1 - Assignment operators shall not be used in expressions that yield a Boolean value
- MISRA C++:2008, 6-2-1 - Assignment operators shall not be used in sub-expressions
- MISRA C:2012, 13.4 - The result of an assignment operator should not be used
- [MITRE, CWE-481](https://cwe.mitre.org/data/definitions/481) - Assigning instead of Comparing
- [CERT, EXP45-C.](https://wiki.sei.cmu.edu/confluence/x/ZNYxBQ) - Do not perform assignments in selection statements
- [CERT, EXP51-J.](https://wiki.sei.cmu.edu/confluence/x/ITZGBQ) - Do not perform assignments in conditional expressions

Reply:
Thanks.